src/RuleMatcher.cc

```diff
@@ -850,13 +850,12 @@ void RuleMatcher::Match(RuleEndpointState* state, Rule::PatternType type, const
         Rule* r = Rule::rule_table[aidx - 1];
 
         if ( AllRulePatternsMatched(r, mpos, accepted_matches) )
-            rule_matches.insert(make_pair(r, mpos));
+            rule_matches[r->Index()] = make_pair(r, mpos);
```

Comment:
This is subtly different in that `insert()` didn't override existing entries, if it doesn't matter, that's probably okay.

Reply:
I tried that but it complains about storing a reference to a pointer, so I settled on that one.

Reply:
Did you try this?

```cpp
            rule_matches.insert({r->Index(), {r, mpos}});
```

The equivalent change on l.668 would have been

```cpp
            rule_matches.insert({r->Index(), r});
```